extensions/binding/org.eclipse.smarthome.binding.bluetooth/src/main/java/org/eclipse/smarthome/binding/bluetooth/BluetoothUtils.java

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.binding.bluetooth;
+
+/**
+ * Static utility methods that are helpful when dealing with Bluetooth data.
+ *
+ * @author Kai Kreuzer - Initial contribution and API
+ *
+ */
+public class BluetoothUtils {
+
+    // used for hex conversions
+    private final static char[] hexArray = "0123456789ABCDEF".toCharArray();
+
+    // private constructor as we only have static methods
+    private BluetoothUtils() {
+    }
+
+    /**
+     * Converts a byte array into a hex string (in format "01 23 45 67 89 0A BC DE").
+     *
+     * @param bytes the byte array
+     * @return the corresponding hex string
+     */
+    public static String bytesToHex(byte[] bytes) {
```

Comment:
Isn't such a method already present / accessible by the bluetooth binding? IIRC The matthew utils brings such a functionality.
If not or if we want to provide our own implementation, wouldn't it make sense to place it in another package (e.g. core)? I don't think bytesToHex is a BT only utility class but rather unrelated to BT only.
Other bindings parsing incoming bytes would be happy to use that function instead of adding an implementation itself.

Shouldn't the last trailing space at the end of the string removed?

If I understand the code correctly it does not return "01 23 45 67 89 0A BC DE" but "01 23 45 67 89 0A BC DE ".

Reply:
>  IIRC The matthew utils brings such a functionality.

We do not have the mattew libs in the Bluetooth binding. I am not aware of any other place where we have such a method.

In Bluetooth, you have to work extensively with byte arrays, so this seemed to be the most logical place to me in the first step. Yes, there might be a few other bindings that could use it as well, but it isn't very common, I guess.

So if you think it would make sense to have it in smarthome.core that is fine for me and I will move it there.

Reply:
> Shouldn't the last trailing space at the end of the string removed?

Absolutely, I've fixed it.